docs/draft/17-url-data.md

```diff
@@ -0,0 +1,126 @@
+---
+permalink: /draft/17
+title: 17/URL data
+parent: Draft specs
+layout: default
+---
+
+# 17/URL data
+
+> Version: 0.1
+>
+> Status: Draft
+>
+> Authors: Felicio Mununga <felicio@status.im>
+
+## Abstract
+
+This document describes serialization, compression and encoding of data within URLs, so the app links could carry enough information to help users with instant previewing and verifying even prior clicking or taking an action on the content.
+
+## Related scope
+
+### Specs
+
+- See <https://github.com/status-im/specs/pull/159> for URL scheme
+
+### Features
+
+- Onboarding website
+- Link preview
+- Link sharing
+- Deep linking
+- Routing and navigation
+
+### App entities
+
+- Community
+- Channel
+- User
+
+## Requirements
+
+### Product
+
+- Encode data within URL until Waku response is instant and considered reliable enough for link previewing and visiting of the onboarding website. Or until a product roadmap for alternative decentralized storage is finalized.
+- A verify state while loading onboarding website until encoded data is matched against Waku response to mitigate identity attacks
+- Accept only `[A-Za-z0-9_-.\u0020]` character class for textual fields
+- Shortest result possible
+
+## Implementation
+
+### Example
+
+- See <https://github.com/status-im/status-web/pull/345/files>
+
+### Data
+
+#### Fields
+
+```protobuf
+syntax = "proto3";
+
+message Community {
+ string display_name = 1;
+ string description = 2;
+ uint32 members_count = 3;
+ string color = 4;
+}
+
+message Channel {
```

Comment:
Should we add uuid here, instead of it being a path segment separate from the encoded data and its signature?

Reply:
The uuid of the channel?

Reply:
For example `30804ea7-bd66-4d5d-91eb-b2dcfe2515b3`. Channel id are concatenations of uuids and community's public keys.

Reply:
Here https://github.com/status-im/specs/pull/159#discussion_r1111762125 are the proposed changes to the scheme spec.

Reply:
if you keep it outside the signature, it can be controlled potentially by an attacker, not clear what the benefit would be if everything else is there, but that's the kind of things that you want it to be in the signature just to be safe

Reply:
That's why I raised, thanks. Updated the spec too.